test/x-address.test.js

```diff
@@ -0,0 +1,120 @@
+const { encode, decode } = require("./../dist/index");
+const fixtures = require('./fixtures/x-codec-fixtures.json')
+
+let json_x1 = {
+    OwnerCount: 0,
+    Account: "XVXdn5wEVm5G4UhEHWDPqjvdeH361P7BsapL4m2D2XnPSwT",
+    PreviousTxnLgrSeq: 7,
+    LedgerEntryType: "AccountRoot",
+    PreviousTxnID: "DF530FB14C5304852F20080B0A8EEF3A6BDD044F41F4EBBD68B8B321145FE4FF",
+    Flags: 0,
+    Sequence: 1,
+    Balance: "10000000000"
+  }
+
+let json_r1 = {
+    OwnerCount: 0,
+    Account: 'rLs1MzkFWCxTbuAHgjeTZK4fcCDDnf2KRv',
+    PreviousTxnLgrSeq: 7,
+    LedgerEntryType: 'AccountRoot',
+    PreviousTxnID: 'DF530FB14C5304852F20080B0A8EEF3A6BDD044F41F4EBBD68B8B321145FE4FF',
+    Flags: 0,
+    Sequence: 1,
+    Balance: '10000000000',
+    SourceTag: 12345,
+  }
+
+let json_null_x = {
+    "OwnerCount": 0,
+    "Account": "rLs1MzkFWCxTbuAHgjeTZK4fcCDDnf2KRv",
+    "Destination": "rLs1MzkFWCxTbuAHgjeTZK4fcCDDnf2KRv",
+    "Issuer": "XVXdn5wEVm5G4UhEHWDPqjvdeH361P4GETfNyyXGaoqBj71",
+    "PreviousTxnLgrSeq": 7,
+    "LedgerEntryType": "AccountRoot",
+    "PreviousTxnID": "DF530FB14C5304852F20080B0A8EEF3A6BDD044F41F4EBBD68B8B321145FE4FF",
+    "Flags": 0,
+    "Sequence": 1,
+    "Balance": "10000000000"
+}
+
+let json_null_r = {
+    "OwnerCount": 0,
+    "Account": "rLs1MzkFWCxTbuAHgjeTZK4fcCDDnf2KRv",
+    "Destination": "rLs1MzkFWCxTbuAHgjeTZK4fcCDDnf2KRv",
+    "Issuer": "rLs1MzkFWCxTbuAHgjeTZK4fcCDDnf2KRv",
+    "PreviousTxnLgrSeq": 7,
+    "LedgerEntryType": "AccountRoot",
+    "PreviousTxnID": "DF530FB14C5304852F20080B0A8EEF3A6BDD044F41F4EBBD68B8B321145FE4FF",
+    "Flags": 0,
+    "Sequence": 1,
+    "Balance": "10000000000"
+}
+
+let invalid_json_issuer_tagged = {
+    "OwnerCount": 0,
+    "Account": "rLs1MzkFWCxTbuAHgjeTZK4fcCDDnf2KRv",
+    "Destination": "rLs1MzkFWCxTbuAHgjeTZK4fcCDDnf2KRv",
+    "Issuer": "XVXdn5wEVm5G4UhEHWDPqjvdeH361P7BsapL4m2D2XnPSwT",
+    "PreviousTxnLgrSeq": 7,
+    "LedgerEntryType": "AccountRoot",
+    "PreviousTxnID": "DF530FB14C5304852F20080B0A8EEF3A6BDD044F41F4EBBD68B8B321145FE4FF",
+    "Flags": 0,
+    "Sequence": 1,
+    "Balance": "10000000000"
+}
+
+let invalid_json_x_and_tagged = {
+    OwnerCount: 0,
+    Account: 'XVXdn5wEVm5G4UhEHWDPqjvdeH361P7BsapL4m2D2XnPSwT',
+    PreviousTxnLgrSeq: 7,
+    LedgerEntryType: 'AccountRoot',
+    PreviousTxnID: 'DF530FB14C5304852F20080B0A8EEF3A6BDD044F41F4EBBD68B8B321145FE4FF',
+    Flags: 0,
+    Sequence: 1,
+    Balance: '10000000000',
+    SourceTag: 12345,
+  }
+
+describe("X-Address account is equivilent to an r-address w/ SourceTag", () => {
+    let encoded_x = encode(json_x1);
+    let encoded_r = encode(json_r1); 
+    test("Can encode with x-Address", () => {
+        expect(encoded_x).toEqual(encoded_r);
+    })
+
+    test("decoded X-address is object w/ source and tag", () => {
+        let decoded_x = decode(encoded_x);
+        expect(decoded_x).toEqual(json_r1);
+    })
+
+    test("Encoding issuer X-Address w/ null destintion tag", () => {
```

Comment:
Consider installing a spell checker to help catch typos like `destintion`. I've used [this spell checker for VS Code](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker).

Technically, the absence of a tag is more like `undefined` than `null`, which is a separate value in JS (though this might be debatable). Still, I think it makes a little more sense to say that the tag is undefined like so:

```suggestion
    test("Encoding issuer X-Address w/ undefined destination tag", () => {
```